pkg/apps/util/const.go

```diff
@@ -28,12 +75,19 @@ const (
 	PostHookPodSuffix = "hook-post"
 
 	// Used only internally by utils:
+
+	// deploymentStatusReasonAnnotation represents the reason for deployment being in a given state
+	// Used for specifying the reason for cancellation or failure of a deployment
+	deploymentStatusReasonAnnotation = "openshift.io/deployment.status-reason"
```

Comment:
@tnozicka note that this collection is used exclusively by util helpers and nothing else... i thinks we should not make it public but have helpers that allow to use them.